### PR TITLE
Feature/scat 694

### PR DESCRIPTION
**Pull request by ronin1972**
Updates messaging to support current requirements

### CODE REVIEW

File: cat/CaT-service.yaml [hunk 1]
```diff
@@ -2597,24 +2670,91 @@ paths:
       security:
         - OAuth2:
             - buyer
-      summary: Create supplier further info request
-    put:
-      description: Respond to a Request for further information
+      summary: Create a message for an event
```
Comment (adrianmilne) [inline cat/CaT-service.yaml:2673]: Think this summary should be create an attachment for a message

File: cat/CaT-service.yaml [hunk 2]
```diff
@@ -2530,20 +2534,18 @@ paths:
...
     parameters:
       - $ref: '#/components/parameters/ProcurementParam'
       - $ref: '#/components/parameters/EventParam'
-      - $ref: 'https://raw.githubusercontent.com/Crown-Commercial-Service/ccs-api-definitions-common/main/CCS_OpenAPI_Standards/CCS_Standard_API_Params.yaml#/components/parameters/SupplierParam'
-
-
+      - $ref: '#/components/parameters/MessageParam'
     get:
-      description: 'Get a request for further information (includes free text and any files}'
+      description: 'Gets a list of messages for the user relating to this event}'
```
Comment (adrianmilne) [inline cat/CaT-service.yaml:2545]: Looks like a copy-paste from get all messages (think this is just getting a single message)

File: cat/CaT-service.yaml [hunk 3]
```diff
@@ -2566,17 +2568,88 @@ paths:
       security:
         - OAuth2:
             - buyer
-      summary: Request Supplier Further Info
+      summary: Request messages for an event
```
Comment (adrianmilne) [inline cat/CaT-service.yaml:2571]: As above - looks like a copy/paste from list response

File: cat/CaT-service.yaml [hunk 4]
```diff
@@ -2466,17 +2466,21 @@ paths:
...
+      
+  /tenders/projects/{proc-id}/events/{event-id}/messages:
     parameters:
       - $ref: '#/components/parameters/ProcurementParam'
       - $ref: '#/components/parameters/EventParam'
-
+      - $ref: '#/components/parameters/MessageDirectionParam'
+      - $ref: '#/components/parameters/MessageSortParam'
+      - $ref: '#/components/parameters/PageParam'
+      - $ref: '#/components/parameters/PageSizeParam'
```
Comment (adrianmilne) [inline cat/CaT-service.yaml:2477]: These last 4 are also showing up as params for the POST operation - think you can move them to just appear under the `get:` section

**Review by tommyb82 — COMMENTED** [review]
As discussed on Slack, happy to go with the `links` approach, just not sure it represents the simplest way of paginating this API.

File: cat/CaT-service.yaml [hunk 5]
```diff
@@ -2466,17 +2466,21 @@ paths:
...
 
     get:
-      description: 'Get a request for further information (includes free text and any files}'
+      description: 'Gets a list of messages for the user relating to this event}'
```
[the remark this answers is not in the document]
Comment (ronin1972) [reply cat/CaT-service.yaml]: Made fix in latest version, thanks

File: cat/CaT-service.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ronin1972) [reply cat/CaT-service.yaml]: Made fix in latest version, thanks

File: cat/CaT-service.yaml (hunk 4 above)
Comment (ronin1972) [reply cat/CaT-service.yaml]: Made fix in latest version, thanks

File: cat/CaT-service.yaml (hunk 3 above)
Comment (ronin1972) [reply cat/CaT-service.yaml]: Made fix in latest version, thanks

File: cat/CaT-service.yaml (hunk 2 above)
Comment (ronin1972) [reply cat/CaT-service.yaml]: Made fix in latest version, thanks

File: cat/CaT-service.yaml (hunk 1 above)
Comment (ronin1972) [reply cat/CaT-service.yaml]: Made fix in latest version, thanks

**Review by sudhakar-reddy-konda — APPROVED** [review]

**Pull request merged**